python/katana/url.py

```diff
@@ -0,0 +1,61 @@
+from urllib import parse
+
+
+class URL:
+    """
+    URL class that maintains semantics similar to pathlib. This class does not
+    intend to act as a universal class for all path related things! Users can use
+    the class in the following ways:
+
+    - URL("file:///home/") / "a" / "b" => URL("file:///home/a/b")
+    - URL("file:///home/", "a", "b") => URL("file:///home/a/b")
```

Comment:
I'm still not a fan of this second constructor. Why do you want to have it?

Reply:
I was mimicking `pathlib` since they have something similar with how `Path` is constructed.

Reply:
Oh do they? WTF. The python standard library has some really odd stuff in it. My instinct is that good API is more important than exactly paralleling lesser known features of another library. But it's up to you.

Reply:
That's a good point. Less is more